src/gui/vector/qgsattributesformproperties.cpp

```diff
@@ -617,6 +633,12 @@ void QgsAttributesFormProperties::toggleShowAliases( bool checked )
   settingShowAliases->setValue( checked );
   mAvailableWidgetsModel->setShowAliases( checked );
   mFormLayoutModel->setShowAliases( checked );
+
+  // When toggling aliases we need to refresh the filter
+  // so that items are reevaluated immediately. A single
+  // invalidate() leaves out other logic (e.g., expanding
+  // nodes), so we play safe and trigger textChanged()
+  emit mSearchLineEdit->textChanged( mSearchLineEdit->text() );
```

Comment:
Not a big fan of this, could we instead move the [=]{ ... } code in the constructor into a private slot of the class, and here call that instead? This emit text->textChanged( text->text() ) is too hackish ;)

Reply:
Better, indeed!